task8/ServletsOnMaven/src/main/java/WorkWithDB/WorkWithDB.java

```diff
@@ -0,0 +1,316 @@
+            );
+            statementToGetPostById.setString(1, id);
+            ResultSet result = statementToGetPostById.executeQuery();
+            if (!result.next()) {
+                return null;
+            }
+            Post post = new Post(id,
+                    result.getString("username"),
+                    LocalDateTime.parse(result.getString("createdAt").replace(' ', 'T')),
+                    result.getString("description"),
+                    new ArrayList<>(),
+                    new ArrayList<>());
+
+            PreparedStatement statementToGetLikes = conn.prepareStatement(
+                    "select username from likes where idOfPost = ?");
+            statementToGetLikes.setString(1, id);
+            result = statementToGetLikes.executeQuery();
+            while (result.next()) {
+                post.getLikes().add(result.getString("username"));
+            }
+
+            PreparedStatement statementToGetHashTags = conn.prepareStatement(
+                    "select hashTag from hashtags where idOfPost = ?");
+            statementToGetHashTags.setString(1, id);
+            result = statementToGetHashTags.executeQuery();
+            while (result.next()) {
+                post.getHashTags().add(result.getString("hashTag"));
+            }
+            return post;
+
+        } catch(SQLException e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    public static String getLastId(DataSource dataSource) {
+        try (Connection conn = dataSource.getConnection()) {
+            ResultSet result = conn.prepareStatement("SELECT * FROM posts ORDER BY idOfPost DESC LIMIT 1;")
+                    .executeQuery();
+            result.next();
+            return result.getString(1);
+
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return "0";
+    }
+
+    public static void addPost(Post post, DataSource dataSource) {
+        int id = Integer.parseInt(getLastId(dataSource));
+        post.setId(id + 1 + "");
+        try (Connection conn = dataSource.getConnection()) {
+
+            PreparedStatement statementToInsertPost = conn.prepareStatement(
+                    "insert posts values (?, ?, ?, ?)");
+            PreparedStatement statementToInsertHashTag = conn.prepareStatement(
+                    "insert hashtags(hashTag, idOfPost) values (?, ?)");
+            PreparedStatement statementToInsertAuthorsOfLike = conn.prepareStatement(
+                    "insert likes(username, idOfPost) values (?, ?)");
+
+            statementToInsertPost.setString(1, post.getId());
+            statementToInsertPost.setString(2, post.getAuthor());
+            statementToInsertPost.setString(3, post.getDescription());
+            statementToInsertPost.setString(4, post.getCreatedAt().toString());
+            statementToInsertPost.executeUpdate();
+
+            post.getHashTags().forEach(hashTag -> {
+                try {
```

Comment:
Please forever avoid nesting `try` into other `try`s, and most of the time you should avoid  putting try-catch block into the lambda. Separate this logic to another methods